clients/python/src/model_registry/core.py

```diff
@@ -17,30 +21,56 @@ class ModelRegistryAPIClient:
     def __init__(
         self,
         server_address: str,
-        port: int,
-        client_key: str | None = None,
-        server_cert: str | None = None,
+        port: int = 443,
+        user_token: str | None = None,
         custom_ca: str | None = None,
     ):
         """Constructor.
 
         Args:
             server_address: Server address.
-            port: Server port.
-            client_key: The PEM-encoded private key as a byte string.
-            server_cert: The PEM-encoded certificate as a byte string.
-            custom_ca: The PEM-encoded root certificates as a byte string.
+            port: Server port. Defaults to 443.
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to envvar CERT.
         """
-        config = MetadataStoreClientConfig()
-        config.host = server_address
-        config.port = port
-        if client_key is not None:
-            config.ssl_config.client_key = client_key
-        if server_cert is not None:
-            config.ssl_config.server_cert = server_cert
-        if custom_ca is not None:
-            config.ssl_config.custom_ca = custom_ca
-        self._store = MLMDStore(config)
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if not sa_token:
+                msg = "Access token must be provided"
+                raise StoreException(msg)
+            token = Path(sa_token).read_bytes()
+        else:
+            token = user_token
```

Comment:
Thank you for the update @isinyaaa !

Unless I miss something:
- this covers for sensible defaults, or user-provided, when the model registry python client is executed inside of a K8s pod/task of pipeline? 🤔 
- but this would make the `user_token` be _mandatory_ when:
  - running inside K8s Notebooks (unless I missed that `KF_PIPELINES_SA_TOKEN_PATH` is available when KF Notebook runs)
  - not running inside of K8s (I believe this is also why the Robotframework tests have been modified)

Wouldn't make more sense not to raise the exception on line 41, and maybe just emit a log line?

I'm a bit concerned user just using this client locally to connect model registry on vanilla KF would not know what to enter for `user_token`?
Can you clarify further about this please.

Reply:
> Wouldn't make more sense not to raise the exception on line 41, and maybe just emit a log line?

Works for me. I wasn't completely sure of the expected behavior on that case. Will update!